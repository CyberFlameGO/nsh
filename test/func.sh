var1=123
func1() {
    var1=456
}
func1
echo $var1

# local
var2=123
function func2 () {
    local var1=456
}
func2
echo $var2

# FIXME: fix parse error
# recursion
fib() {
    if [ $1 -lt 2 ]; then
        echo $1
        return
    fi
    x=$(( $1 - 1 ))
    y=$(( $1 - 2 ))
    a=$(fib $x)
    b=$(fib $y)
    echo $(( a + b ))
}

fib 7
