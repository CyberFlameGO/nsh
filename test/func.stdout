456
123
13
